build/operator/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM centos:centos7
+FROM quay.io/centos/centos:centos7.9.2009
```

Comment:
imo we don't need to be that strict, we usually pin the major only (for example on CNAO we used centos7 and when we need fedora we are using fedora32)
Moreover in case minor / patch versions are added, we better have them as well, because they can be either security / bug fix,
and in case they are breaking something, we will know and adapt our code, and not only when we change this line.
Changing it also might require maintenance that we would need to bump it manually from time to time

Reply:
We can even use this PR to pin it to ubi8-minimal or tu centos 8 stream, centos7 is quite ancient.

Reply:
One is pinning and the other is upgrading. I do not think it is the same, one has zero risk and the other requires much more work and has potential to find numerous issues.

Reply:
I agree with @EdDev.
The replacement of a base image is a subject for another PR.

Reply:
Then let's just pin major.minor so we receive patch fixes.

```
quay.io/centos/centos:centos7.9
```

Reply:
So you assume the patches do not cause any regression, right?
As I see it, it is still not under control... it will update the version per-build without any control.

I find this problematic, but it may worth checking how other project manage such things.
If I take golang as an example, they pin it with a specific hash. Which allows predictable builds without surprises.

I guess a policy needs to be define in general, something that all parts of the project can follow.

Reply:
IMO builds should be reproducible.
For example, if you don't pin the base image version and you fix a bug in an older release and rebuild the image - you get new changes from the base image that weren't there when you've first released the image.

Its a trade-off between using known and tested packages vs. the cost of maintenance / automation.